Navigate TurtleBot3 Issue

I'm trying the navigation demo and cannot get the navigation system work properly. It's my first time using Webots so I just follow the instruction and install everything by package manager. System is Ubuntu 20 with ROS foxy. Webots is R2021b and is downloaded automatically.

The webots and rviz gui show up, but there's no map topic and no transform from /odom to /map/

From log what I can see it fails to load plugin `webots_ros2_driver::Ros2IMU`(whose .so file actually exists) and then `/controller_manager` fails to run causing `spawner_diffdrive_controller` & `spawner_joint_state_broadcaster`'s failure.

Detailed log is listed below.

Terminal output of webots:
```
[INFO] [launch]: All log files can be found below /home/mhbc/.ros/log/2021-08-10-11-51-01-246711-mhbc-ROS-25904
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [spawner.py-1]: process started with pid [25908]
[INFO] [spawner.py-2]: process started with pid [25910]
[INFO] [turtlebot3_burger_example.wbt" --batch --mode=realtime-3]: process started with pid [25913]
[INFO] [robot_state_publisher-4]: process started with pid [25915]
[INFO] [driver-5]: process started with pid [25919]
[INFO] [static_transform_publisher-6]: process started with pid [25921]
[driver-5] Cannot open file: /tmp/webots-25918-2iixXa/WEBOTS_SERVER (retry count 1)
[static_transform_publisher-6] [INFO] [1628567461.333937935] [static_transform_publisher_2vigRTd6kwzljIGM]: Spinning until killed publishing transform from 'base_link' to 'base_footprint'
[robot_state_publisher-4] Parsing robot urdf xml string.
[robot_state_publisher-4] [INFO] [1628567461.335604475] [robot_state_publisher]: got segment 
[driver-5] Cannot open file: /tmp/webots-25918-2iixXa/WEBOTS_SERVER (retry count 2)
[driver-5] Cannot open file: /tmp/webots-25918-2iixXa/WEBOTS_SERVER (retry count 3)
[driver-5] Cannot open file: /tmp/webots-25918-2iixXa/WEBOTS_SERVER (retry count 4)
[robot_state_publisher-4] Parsing robot urdf xml string.
[robot_state_publisher-4] Link LDS-01 had 6 children
[robot_state_publisher-4] Link solid(1) had 0 children
[robot_state_publisher-4] Link screw0 had 0 children
[robot_state_publisher-4] Link screw1 had 0 children
[robot_state_publisher-4] Link screw2 had 0 children
[robot_state_publisher-4] Link screw3 had 0 children
[robot_state_publisher-4] Link solid_0 had 0 children
[robot_state_publisher-4] Link accelerometer had 0 children
[robot_state_publisher-4] Link compass had 0 children
[robot_state_publisher-4] Link gyro had 0 children
[robot_state_publisher-4] Link imu_link had 1 children
[robot_state_publisher-4] Link solid had 0 children
[robot_state_publisher-4] Link inertial_unit had 0 children
[robot_state_publisher-4] Link left wheel had 0 children
[robot_state_publisher-4] Link right wheel had 0 children
[robot_state_publisher-4] [INFO] [1628567465.681053929] [robot_state_publisher]: got segment LDS-01
[robot_state_publisher-4] [INFO] [1628567465.681093664] [robot_state_publisher]: got segment accelerometer
[robot_state_publisher-4] [INFO] [1628567465.681101769] [robot_state_publisher]: got segment base_link
[robot_state_publisher-4] [INFO] [1628567465.681108703] [robot_state_publisher]: got segment compass
[robot_state_publisher-4] [INFO] [1628567465.681115355] [robot_state_publisher]: got segment gyro
[robot_state_publisher-4] [INFO] [1628567465.681122278] [robot_state_publisher]: got segment imu_link
[robot_state_publisher-4] [INFO] [1628567465.681128921] [robot_state_publisher]: got segment inertial_unit
[robot_state_publisher-4] [INFO] [1628567465.681135443] [robot_state_publisher]: got segment left wheel
[robot_state_publisher-4] [INFO] [1628567465.681142105] [robot_state_publisher]: got segment right wheel
[robot_state_publisher-4] [INFO] [1628567465.681148638] [robot_state_publisher]: got segment screw0
[robot_state_publisher-4] [INFO] [1628567465.681155120] [robot_state_publisher]: got segment screw1
[robot_state_publisher-4] [INFO] [1628567465.681161562] [robot_state_publisher]: got segment screw2
[robot_state_publisher-4] [INFO] [1628567465.681168024] [robot_state_publisher]: got segment screw3
[robot_state_publisher-4] [INFO] [1628567465.681174637] [robot_state_publisher]: got segment solid
[robot_state_publisher-4] [INFO] [1628567465.681181089] [robot_state_publisher]: got segment solid(1)
[robot_state_publisher-4] [INFO] [1628567465.681189164] [robot_state_publisher]: got segment solid_0
[driver-5] terminate called after throwing an instance of 'pluginlib::LibraryLoadException'
[driver-5]   what():  Could not find library corresponding to plugin webots_ros2_driver::Ros2IMU. Make sure the plugin description XML file has the correct name of the library and that the library actually exists.
[ERROR] [driver-5]: process has died [pid 25919, exit code -6, cmd '/opt/ros/foxy/lib/webots_ros2_driver/driver --ros-args --params-file /tmp/launch_params_si1wvkcb --params-file /opt/ros/foxy/share/webots_ros2_turtlebot/resource/ros2control.yml -r /diffdrive_controller/cmd_vel_unstamped:=/cmd_vel'].
[spawner.py-2] [INFO] [1628567471.443953622] [spawner_diffdrive_controller]: Waiting for /controller_manager services
[spawner.py-1] [INFO] [1628567471.450646803] [spawner_joint_state_broadcaster]: Waiting for /controller_manager services
[spawner.py-2] [INFO] [1628567473.451271959] [spawner_diffdrive_controller]: Waiting for /controller_manager services
[spawner.py-1] [INFO] [1628567473.458582122] [spawner_joint_state_broadcaster]: Waiting for /controller_manager services
[spawner.py-2] [INFO] [1628567475.458691158] [spawner_diffdrive_controller]: Waiting for /controller_manager services
[spawner.py-1] [INFO] [1628567475.465675046] [spawner_joint_state_broadcaster]: Waiting for /controller_manager services
[spawner.py-2] [INFO] [1628567477.466440402] [spawner_diffdrive_controller]: Waiting for /controller_manager services
[spawner.py-1] [INFO] [1628567477.472865536] [spawner_joint_state_broadcaster]: Waiting for /controller_manager services
[spawner.py-2] [INFO] [1628567479.473855647] [spawner_diffdrive_controller]: Waiting for /controller_manager services
[spawner.py-1] [INFO] [1628567479.480600205] [spawner_joint_state_broadcaster]: Waiting for /controller_manager services
[spawner.py-2] [ERROR] [1628567481.481165086] [spawner_diffdrive_controller]: Controller manager not available
[spawner.py-1] [ERROR] [1628567481.487731267] [spawner_joint_state_broadcaster]: Controller manager not available
[ERROR] [spawner.py-2]: process has died [pid 25910, exit code 1, cmd 'bash -c sleep 10; $0 $@ /opt/ros/foxy/lib/controller_manager/spawner.py diffdrive_controller --ros-args'].
[ERROR] [spawner.py-1]: process has died [pid 25908, exit code 1, cmd 'bash -c sleep 10; $0 $@ /opt/ros/foxy/lib/controller_manager/spawner.py joint_state_broadcaster --ros-args'].
^C[WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
[robot_state_publisher-4] [INFO] [1628567509.820492625] [rclcpp]: signal_handler(signal_value=2)
[static_transform_publisher-6] [INFO] [1628567509.820502623] [rclcpp]: signal_handler(signal_value=2)
[INFO] [static_transform_publisher-6]: process has finished cleanly [pid 25921]
[INFO] [robot_state_publisher-4]: process has finished cleanly [pid 25915]
[ERROR] [turtlebot3_burger_example.wbt" --batch --mode=realtime-3]: process has died [pid 25913, exit code -2, cmd '"/home/mhbc/.ros/webotsR2021b/webots/webots" "/opt/ros/foxy/share/webots_ros2_turtlebot/worlds/turtlebot3_burger_example.wbt" --batch --mode=realtime'].
```

Terminal output of navigation2:
```
[INFO] [launch]: All log files can be found below /home/mhbc/.ros/log/2021-08-10-11-51-21-272914-mhbc-ROS-26112
[INFO] [launch]: Default logging verbosity is set to INFO
/opt/ros/foxy/share/turtlebot3_navigation2/launch/navigation2.launch.py:78: UserWarning: The parameter 'node_executable' is deprecated, use 'executable' instead
  Node(
/opt/ros/foxy/share/turtlebot3_navigation2/launch/navigation2.launch.py:78: UserWarning: The parameter 'node_name' is deprecated, use 'name' instead
  Node(
[INFO] [map_server-1]: process started with pid [26133]
[INFO] [amcl-2]: process started with pid [26135]
[INFO] [lifecycle_manager-3]: process started with pid [26137]
[INFO] [controller_server-4]: process started with pid [26139]
[INFO] [planner_server-5]: process started with pid [26141]
[INFO] [recoveries_server-6]: process started with pid [26143]
[INFO] [bt_navigator-7]: process started with pid [26145]
[INFO] [waypoint_follower-8]: process started with pid [26147]
[INFO] [lifecycle_manager-9]: process started with pid [26149]
[INFO] [rviz2-10]: process started with pid [26151]
[lifecycle_manager-3] [INFO] [1628567481.570166270] [lifecycle_manager_localization]: Creating
[lifecycle_manager-3] [INFO] [1628567481.575101178] [lifecycle_manager_localization]: Creating and initializing lifecycle service clients
[lifecycle_manager-3] [INFO] [1628567481.578319902] [lifecycle_manager_localization]: Starting managed nodes bringup...
[lifecycle_manager-3] [INFO] [1628567481.578357693] [lifecycle_manager_localization]: Configuring map_server
[lifecycle_manager-9] [INFO] [1628567481.582730622] [lifecycle_manager_navigation]: Creating
[recoveries_server-6] [INFO] [1628567481.585725354] [recoveries_server]: 
[recoveries_server-6] 	recoveries_server lifecycle node launched. 
[recoveries_server-6] 	Waiting on external lifecycle transitions to activate
[recoveries_server-6] 	See https://design.ros2.org/articles/node_lifecycle.html for more information.
[lifecycle_manager-9] [INFO] [1628567481.588068929] [lifecycle_manager_navigation]: Creating and initializing lifecycle service clients
[bt_navigator-7] [INFO] [1628567481.588813021] [bt_navigator]: 
[bt_navigator-7] 	bt_navigator lifecycle node launched. 
[bt_navigator-7] 	Waiting on external lifecycle transitions to activate
[bt_navigator-7] 	See https://design.ros2.org/articles/node_lifecycle.html for more information.
[bt_navigator-7] [INFO] [1628567481.589068062] [bt_navigator]: Creating
[map_server-1] [INFO] [1628567481.590896036] [map_server]: 
[map_server-1] 	map_server lifecycle node launched. 
[map_server-1] 	Waiting on external lifecycle transitions to activate
[map_server-1] 	See https://design.ros2.org/articles/node_lifecycle.html for more information.
[map_server-1] [INFO] [1628567481.590996024] [map_server]: Creating
[planner_server-5] [INFO] [1628567481.591560768] [planner_server]: 
[planner_server-5] 	planner_server lifecycle node launched. 
[planner_server-5] 	Waiting on external lifecycle transitions to activate
[planner_server-5] 	See https://design.ros2.org/articles/node_lifecycle.html for more information.
[controller_server-4] [INFO] [1628567481.592076751] [controller_server]: 
[controller_server-4] 	controller_server lifecycle node launched. 
[controller_server-4] 	Waiting on external lifecycle transitions to activate
[controller_server-4] 	See https://design.ros2.org/articles/node_lifecycle.html for more information.
[planner_server-5] [INFO] [1628567481.592287287] [planner_server]: Creating
[map_server-1] [INFO] [1628567481.593154571] [map_server]: Configuring
[map_server-1] [INFO] [map_io]: Loading yaml file: /opt/ros/foxy/share/webots_ros2_turtlebot/resource/turtlebot3_burger_example_map.yaml
[map_server-1] [DEBUG] [map_io]: resolution: 0.05
[map_server-1] [DEBUG] [map_io]: origin[0]: -2.06
[map_server-1] [DEBUG] [map_io]: origin[1]: -6.84
[map_server-1] [DEBUG] [map_io]: origin[2]: 0
[map_server-1] [DEBUG] [map_io]: free_thresh: 0.25
[map_server-1] [DEBUG] [map_io]: occupied_thresh: 0.65
[map_server-1] [DEBUG] [map_io]: mode: trinary
[map_server-1] [DEBUG] [map_io]: negate: 0
[map_server-1] [INFO] [map_io]: Loading image_file: /opt/ros/foxy/share/webots_ros2_turtlebot/resource/turtlebot3_burger_example_map.pgm
[controller_server-4] [INFO] [1628567481.594024390] [controller_server]: Creating controller server
[lifecycle_manager-9] [INFO] [1628567481.594074665] [lifecycle_manager_navigation]: Starting managed nodes bringup...
[lifecycle_manager-9] [INFO] [1628567481.594102648] [lifecycle_manager_navigation]: Configuring controller_server
[map_server-1] [DEBUG] [map_io]: Read map /opt/ros/foxy/share/webots_ros2_turtlebot/resource/turtlebot3_burger_example_map.pgm: 164 X 223 map @ 0.05 m/cell
[amcl-2] [INFO] [1628567481.600771042] [amcl]: 
[amcl-2] 	amcl lifecycle node launched. 
[amcl-2] 	Waiting on external lifecycle transitions to activate
[amcl-2] 	See https://design.ros2.org/articles/node_lifecycle.html for more information.
[waypoint_follower-8] [INFO] [1628567481.600851183] [waypoint_follower]: 
[waypoint_follower-8] 	waypoint_follower lifecycle node launched. 
[waypoint_follower-8] 	Waiting on external lifecycle transitions to activate
[waypoint_follower-8] 	See https://design.ros2.org/articles/node_lifecycle.html for more information.
[waypoint_follower-8] [INFO] [1628567481.600951091] [waypoint_follower]: Creating
[amcl-2] [INFO] [1628567481.600910144] [amcl]: Creating
[lifecycle_manager-3] [INFO] [1628567481.601451083] [lifecycle_manager_localization]: Configuring amcl
[amcl-2] [INFO] [1628567481.601618999] [amcl]: Configuring
[amcl-2] [INFO] [1628567481.601673612] [amcl]: initTransforms
[amcl-2] [INFO] [1628567481.605684418] [amcl]: initPubSub
[amcl-2] [INFO] [1628567481.607788513] [amcl]: Subscribed to map topic.
[lifecycle_manager-3] [INFO] [1628567481.610766544] [lifecycle_manager_localization]: Activating map_server
[map_server-1] [INFO] [1628567481.610874196] [map_server]: Activating
[amcl-2] [INFO] [1628567481.611130169] [amcl]: Received a 164 X 223 map @ 0.050 m/pix
[controller_server-4] [INFO] [1628567481.611168842] [local_costmap.local_costmap]: 
[controller_server-4] 	local_costmap lifecycle node launched. 
[controller_server-4] 	Waiting on external lifecycle transitions to activate
[controller_server-4] 	See https://design.ros2.org/articles/node_lifecycle.html for more information.
[lifecycle_manager-3] [INFO] [1628567481.611296672] [lifecycle_manager_localization]: Activating amcl
[controller_server-4] [INFO] [1628567481.611410817] [local_costmap.local_costmap]: Creating Costmap
[amcl-2] [INFO] [1628567481.611834335] [amcl]: Activating
[amcl-2] [WARN] [1628567481.611850486] [amcl]: Publishing the particle cloud as geometry_msgs/PoseArray msg is deprecated, will be published as nav2_msgs/ParticleCloud in the future
[lifecycle_manager-3] [INFO] [1628567481.612083565] [lifecycle_manager_localization]: Managed nodes are active
[planner_server-5] [INFO] [1628567481.613330374] [global_costmap.global_costmap]: 
[planner_server-5] 	global_costmap lifecycle node launched. 
[planner_server-5] 	Waiting on external lifecycle transitions to activate
[planner_server-5] 	See https://design.ros2.org/articles/node_lifecycle.html for more information.
[planner_server-5] [INFO] [1628567481.613550429] [global_costmap.global_costmap]: Creating Costmap
[controller_server-4] [INFO] [1628567481.616914006] [controller_server]: Configuring controller interface
[controller_server-4] [INFO] [1628567481.617055843] [controller_server]: Controller frequency set to 20.0000Hz
[controller_server-4] [INFO] [1628567481.617075019] [local_costmap.local_costmap]: Configuring
[controller_server-4] [INFO] [1628567481.619654679] [local_costmap.local_costmap]: Using plugin "voxel_layer"
[controller_server-4] [INFO] [1628567481.622038861] [local_costmap.local_costmap]: Subscribed to Topics: scan
[controller_server-4] [INFO] [1628567481.624822596] [local_costmap.local_costmap]: Initialized plugin "voxel_layer"
[controller_server-4] [INFO] [1628567481.624846792] [local_costmap.local_costmap]: Using plugin "inflation_layer"
[controller_server-4] [INFO] [1628567481.625364687] [local_costmap.local_costmap]: Initialized plugin "inflation_layer"
[controller_server-4] [INFO] [1628567481.629530336] [controller_server]: Created progress_checker : progress_checker of type nav2_controller::SimpleProgressChecker
[controller_server-4] [INFO] [1628567481.629789484] [controller_server]: Created goal_checker : goal_checker of type nav2_controller::SimpleGoalChecker
[controller_server-4] [INFO] [1628567481.631462075] [controller_server]: Created controller : FollowPath of type dwb_core::DWBLocalPlanner
[controller_server-4] [INFO] [1628567481.631926760] [controller_server]: Setting transform_tolerance to 0.200000
[controller_server-4] [INFO] [1628567481.639993900] [controller_server]: Using critic "RotateToGoal" (dwb_critics::RotateToGoalCritic)
[controller_server-4] [INFO] [1628567481.640780442] [controller_server]: Critic plugin initialized
[controller_server-4] [INFO] [1628567481.641082811] [controller_server]: Using critic "Oscillation" (dwb_critics::OscillationCritic)
[controller_server-4] [INFO] [1628567481.641295973] [controller_server]: Critic plugin initialized
[controller_server-4] [INFO] [1628567481.641337291] [controller_server]: Using critic "BaseObstacle" (dwb_critics::BaseObstacleCritic)
[controller_server-4] [INFO] [1628567481.641377196] [controller_server]: Critic plugin initialized
[controller_server-4] [INFO] [1628567481.641408325] [controller_server]: Using critic "GoalAlign" (dwb_critics::GoalAlignCritic)
[controller_server-4] [INFO] [1628567481.641487474] [controller_server]: Critic plugin initialized
[controller_server-4] [INFO] [1628567481.641515497] [controller_server]: Using critic "PathAlign" (dwb_critics::PathAlignCritic)
[controller_server-4] [INFO] [1628567481.641572173] [controller_server]: Critic plugin initialized
[controller_server-4] [INFO] [1628567481.641600517] [controller_server]: Using critic "PathDist" (dwb_critics::PathDistCritic)
[controller_server-4] [INFO] [1628567481.641651924] [controller_server]: Critic plugin initialized
[controller_server-4] [INFO] [1628567481.641679826] [controller_server]: Using critic "GoalDist" (dwb_critics::GoalDistCritic)
[controller_server-4] [INFO] [1628567481.641723228] [controller_server]: Critic plugin initialized
[controller_server-4] [INFO] [1628567481.641734459] [controller_server]: Controller Server has FollowPath  controllers available.
[lifecycle_manager-9] [INFO] [1628567481.646891185] [lifecycle_manager_navigation]: Configuring planner_server
[planner_server-5] [INFO] [1628567481.647078017] [planner_server]: Configuring
[planner_server-5] [INFO] [1628567481.647100850] [global_costmap.global_costmap]: Configuring
[planner_server-5] [INFO] [1628567481.649546908] [global_costmap.global_costmap]: Using plugin "static_layer"
[planner_server-5] [INFO] [1628567481.651387596] [global_costmap.global_costmap]: Subscribing to the map topic (/map) with transient local durability
[planner_server-5] [INFO] [1628567481.652245052] [global_costmap.global_costmap]: Initialized plugin "static_layer"
[planner_server-5] [INFO] [1628567481.652260531] [global_costmap.global_costmap]: Using plugin "obstacle_layer"
[planner_server-5] [INFO] [1628567481.652827409] [global_costmap.global_costmap]: Subscribed to Topics: scan
[planner_server-5] [INFO] [1628567481.654885316] [global_costmap.global_costmap]: Initialized plugin "obstacle_layer"
[planner_server-5] [INFO] [1628567481.654904292] [global_costmap.global_costmap]: Using plugin "inflation_layer"
[planner_server-5] [INFO] [1628567481.655151618] [global_costmap.global_costmap]: Initialized plugin "inflation_layer"
[planner_server-5] [INFO] [1628567481.659046146] [planner_server]: Created global planner plugin GridBased of type nav2_navfn_planner/NavfnPlanner
[planner_server-5] [INFO] [1628567481.659079609] [planner_server]: Configuring plugin GridBased of type NavfnPlanner
[planner_server-5] [INFO] [1628567481.662816830] [global_costmap.global_costmap]: StaticLayer: Resizing costmap to 164 X 223 at 0.050000 m/pix
[planner_server-5] [INFO] [1628567481.663368038] [planner_server]: Planner Server has GridBased  planners available.
[lifecycle_manager-9] [INFO] [1628567481.666198220] [lifecycle_manager_navigation]: Configuring recoveries_server
[recoveries_server-6] [INFO] [1628567481.666369693] [recoveries_server]: Configuring
[recoveries_server-6] [INFO] [1628567481.669259137] [recoveries_server]: Creating recovery plugin spin of type nav2_recoveries/Spin
[recoveries_server-6] [INFO] [1628567481.670262838] [recoveries_server]: Configuring spin
[recoveries_server-6] [INFO] [1628567481.672730838] [recoveries_server]: Creating recovery plugin back_up of type nav2_recoveries/BackUp
[recoveries_server-6] [INFO] [1628567481.673424475] [recoveries_server]: Configuring back_up
[recoveries_server-6] [INFO] [1628567481.675386351] [recoveries_server]: Creating recovery plugin wait of type nav2_recoveries/Wait
[recoveries_server-6] [INFO] [1628567481.676306525] [recoveries_server]: Configuring wait
[lifecycle_manager-9] [INFO] [1628567481.679432494] [lifecycle_manager_navigation]: Configuring bt_navigator
[bt_navigator-7] [INFO] [1628567481.679629055] [bt_navigator]: Configuring
[bt_navigator-7] [INFO] [1628567481.694864860] [bt_navigator_rclcpp_node]: Waiting for "compute_path_to_pose" action server
[bt_navigator-7] [INFO] [1628567481.695027617] [bt_navigator_rclcpp_node]: "ComputePathToPose" BtActionNode initialized
[bt_navigator-7] [INFO] [1628567481.695687610] [bt_navigator_rclcpp_node]: Waiting for "global_costmap/clear_entirely_global_costmap" service
[bt_navigator-7] [INFO] [1628567481.695755278] [bt_navigator_rclcpp_node]: "ClearGlobalCostmap-Context" BtServiceNode initialized
[bt_navigator-7] [INFO] [1628567481.697753462] [bt_navigator_rclcpp_node]: Waiting for "follow_path" action server
[bt_navigator-7] [INFO] [1628567481.697849533] [bt_navigator_rclcpp_node]: "FollowPath" BtActionNode initialized
[bt_navigator-7] [INFO] [1628567481.698480322] [bt_navigator_rclcpp_node]: Waiting for "local_costmap/clear_entirely_local_costmap" service
[bt_navigator-7] [INFO] [1628567481.698533382] [bt_navigator_rclcpp_node]: "ClearLocalCostmap-Context" BtServiceNode initialized
[bt_navigator-7] [INFO] [1628567481.699071175] [bt_navigator_rclcpp_node]: Waiting for "local_costmap/clear_entirely_local_costmap" service
[bt_navigator-7] [INFO] [1628567481.699119917] [bt_navigator_rclcpp_node]: "ClearLocalCostmap-Subtree" BtServiceNode initialized
[bt_navigator-7] [INFO] [1628567481.699618627] [bt_navigator_rclcpp_node]: Waiting for "global_costmap/clear_entirely_global_costmap" service
[bt_navigator-7] [INFO] [1628567481.699679431] [bt_navigator_rclcpp_node]: "ClearGlobalCostmap-Subtree" BtServiceNode initialized
[bt_navigator-7] [INFO] [1628567481.701821627] [bt_navigator_rclcpp_node]: Waiting for "spin" action server
[bt_navigator-7] [INFO] [1628567481.701983943] [bt_navigator_rclcpp_node]: "Spin" BtActionNode initialized
[bt_navigator-7] [INFO] [1628567481.703955929] [bt_navigator_rclcpp_node]: Waiting for "wait" action server
[bt_navigator-7] [INFO] [1628567481.704050938] [bt_navigator_rclcpp_node]: "Wait" BtActionNode initialized
[lifecycle_manager-9] [INFO] [1628567481.704979477] [lifecycle_manager_navigation]: Configuring waypoint_follower
[waypoint_follower-8] [INFO] [1628567481.705148646] [waypoint_follower]: Configuring
[lifecycle_manager-9] [INFO] [1628567481.714024538] [lifecycle_manager_navigation]: Activating controller_server
[controller_server-4] [INFO] [1628567481.714140777] [controller_server]: Activating
[controller_server-4] [INFO] [1628567481.714170774] [local_costmap.local_costmap]: Activating
[controller_server-4] [INFO] [1628567481.714183979] [local_costmap.local_costmap]: Checking transform
[controller_server-4] [INFO] [1628567481.714219536] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567482.214273409] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567482.431137283] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-10] [INFO] [1628567482.431297956] [rviz2]: OpenGl version: 4.6 (GLSL 4.6)
[rviz2-10] [INFO] [1628567482.541925705] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-10] [WARN] [1628567482.621662764] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[rviz2-10] [WARN] [1628567482.626489548] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[rviz2-10] [WARN] [1628567482.632026069] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
[rviz2-10] [INFO] [1628567482.644783114] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567482.644962162] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[rviz2-10] [INFO] [1628567482.703147116] [rviz2]: Trying to create a map of size 164 x 223 using 1 swatches
[controller_server-4] [INFO] [1628567482.714266966] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567483.214269780] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567483.645185810] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567483.645332607] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567483.714268686] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567484.214278231] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567484.645492090] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567484.645660227] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567484.714273497] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567485.214263723] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567485.645821609] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567485.646001718] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567485.714270520] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567486.214286061] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567486.646217669] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567486.646375937] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567486.714275142] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567487.214278779] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567487.646581835] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567487.646716859] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567487.714268168] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567488.214272625] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567488.646868380] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567488.647041205] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567488.714270227] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567489.214269832] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567489.647207309] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567489.647370186] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567489.714267683] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567490.214271254] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567490.647595026] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567490.647773292] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567490.714275284] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567491.214268453] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567491.647949046] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567491.648098437] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567491.714272070] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567492.214274724] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567492.648308301] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567492.648474214] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567492.714273550] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567493.214268328] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567493.648699451] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567493.648863069] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567493.714271740] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567494.214275523] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567494.649084406] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567494.649255218] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567494.714279735] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567495.214283134] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567495.649424893] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567495.649622204] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567495.714273317] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[amcl-2] [INFO] [1628567495.819367295] [amcl]: initialPoseReceived
[amcl-2] [INFO] [1628567495.819512608] [amcl]: Setting pose (0.000000): 0.000 0.000 0.000
[controller_server-4] [INFO] [1628567496.214269601] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567496.649851788] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567496.650020656] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567496.714271445] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567497.214268708] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567497.650238002] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567497.650395910] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567497.714272713] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567498.214273051] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567498.650552958] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567498.650708882] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567498.714268057] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567499.214272500] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567499.650877237] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567499.651048299] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567499.714261763] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567500.214275902] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567500.651217532] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567500.651387122] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567500.714271224] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567501.214274000] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567501.651631161] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567501.651806501] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567501.714277576] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567502.214277304] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567502.652029127] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567502.652197184] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567502.714276168] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567503.214270384] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567503.652359941] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567503.652524311] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567503.714268625] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567504.214270102] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567504.652731909] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567504.652885258] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567504.714271117] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567505.214271049] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[rviz2-10] [INFO] [1628567505.653041115] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] [INFO] [1628567505.653212558] [rviz2]: Sending lifecycle_manager_navigation/is_active request
[controller_server-4] [INFO] [1628567505.714272673] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
[controller_server-4] [INFO] [1628567506.214273554] [local_costmap.local_costmap]: Timed out waiting for transform from base_link to odom to become available, tf error: Invalid frame ID "odom" passed to canTransform argument target_frame - frame does not exist
^C[WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
[rviz2-10] [INFO] [1628567506.467149893] [rclcpp]: signal_handler(signal_value=2)
[rviz2-10] [INFO] [1628567506.467671696] [rviz2]: Waiting for the lifecycle_manager_navigation/is_active service...
[rviz2-10] 
[rviz2-10] >>> [rcutils|error_handling.c:108] rcutils_set_error_state()
[rviz2-10] This error state is being overwritten:
[rviz2-10] 
[rviz2-10]   'rcl node's context is invalid, at /tmp/binarydeb/ros-foxy-rcl-1.1.11/src/rcl/node.c:441'
[rviz2-10] 
[rviz2-10] with this new error message:
[rviz2-10] 
[rviz2-10]   'the given context is not valid, either rcl_init() was not called or rcl_shutdown() was called., at /tmp/binarydeb/ros-foxy-rcl-1.1.11/src/rcl/wait.c:130'
[rviz2-10] 
[rviz2-10] rcutils_reset_error() should be called after error handling to avoid this.
[rviz2-10] <<<
[rviz2-10] terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
[rviz2-10]   what():  failed to initialize wait set: the given context is not valid, either rcl_init() was not called or rcl_shutdown() was called., at /tmp/binarydeb/ros-foxy-rcl-1.1.11/src/rcl/wait.c:130
[bt_navigator-7] [INFO] [1628567506.467154822] [rclcpp]: signal_handler(signal_value=2)
[bt_navigator-7] [INFO] [1628567506.470035439] [bt_navigator]: Destroying
[INFO] [recoveries_server-6]: process has finished cleanly [pid 26143]
[INFO] [controller_server-4]: process has finished cleanly [pid 26139]
[lifecycle_manager-9] [INFO] [1628567506.467149963] [rclcpp]: signal_handler(signal_value=2)
[lifecycle_manager-9] [ERROR] [1628567506.467337987] [lifecycle_manager_navigation]: Failed to change state for node: controller_server
[lifecycle_manager-9] [ERROR] [1628567506.467349970] [lifecycle_manager_navigation]: Failed to bring up all requested nodes. Aborting bringup.
[lifecycle_manager-9] terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
[lifecycle_manager-9]   what():  Failed to create interrupt guard condition in Executor constructor: the given context is not valid, either rcl_init() was not called or rcl_shutdown() was called., at /tmp/binarydeb/ros-foxy-rcl-1.1.11/src/rcl/guard_condition.c:67
[INFO] [planner_server-5]: process has finished cleanly [pid 26141]
[waypoint_follower-8] [INFO] [1628567506.467160623] [rclcpp]: signal_handler(signal_value=2)
[waypoint_follower-8] [INFO] [1628567506.468889710] [waypoint_follower]: Destroying
[amcl-2] [INFO] [1628567506.467169009] [rclcpp]: signal_handler(signal_value=2)
[amcl-2] [INFO] [1628567506.471245308] [amcl]: Destroying
[recoveries_server-6] [INFO] [1628567506.467167235] [rclcpp]: signal_handler(signal_value=2)
[planner_server-5] [INFO] [1628567506.467168658] [rclcpp]: signal_handler(signal_value=2)
[lifecycle_manager-3] [INFO] [1628567506.467169309] [rclcpp]: signal_handler(signal_value=2)
[controller_server-4] [INFO] [1628567506.467172305] [rclcpp]: signal_handler(signal_value=2)
[controller_server-4] [INFO] [1628567506.467384925] [local_costmap.local_costmap]: start
[controller_server-4] [ERROR] [1628567506.467584231] []: Caught exception in callback for transition 13
[controller_server-4] [ERROR] [1628567506.467598167] []: Original error: could not create subscription: rcl node's context is invalid, at /tmp/binarydeb/ros-foxy-rcl-1.1.11/src/rcl/node.c:441
[controller_server-4] [ERROR] [1628567506.467621311] []: Failed to finish transition 3. Current state is now: errorprocessing (Could not publish transition: publisher's context is invalid, at /tmp/binarydeb/ros-foxy-rcl-1.1.11/src/rcl/publisher.c:423, at /tmp/binarydeb/ros-foxy-rcl-lifecycle-1.1.11/src/rcl_lifecycle.c:367)
[map_server-1] [INFO] [1628567506.467177835] [rclcpp]: signal_handler(signal_value=2)
[map_server-1] [INFO] [1628567506.468282136] [map_server]: Destroying
[bt_navigator-7] [PublisherZMQ] Server quitting.
[bt_navigator-7] [PublisherZMQ] just died. Exeption Context was terminated
[bt_navigator-7] [PublisherZMQ] Publisher quitting.
[bt_navigator-7] [PublisherZMQ] just died. Exeption Context was terminated
[lifecycle_manager-3] [INFO] [1628567506.476703310] [lifecycle_manager_localization]: Destroying
[INFO] [map_server-1]: process has finished cleanly [pid 26133]
[INFO] [waypoint_follower-8]: process has finished cleanly [pid 26147]
[INFO] [lifecycle_manager-3]: process has finished cleanly [pid 26137]
[INFO] [amcl-2]: process has finished cleanly [pid 26135]
[INFO] [bt_navigator-7]: process has finished cleanly [pid 26145]
[ERROR] [rviz2-10]: process has died [pid 26151, exit code -6, cmd '/opt/ros/foxy/lib/rviz2/rviz2 -d /opt/ros/foxy/share/nav2_bringup/rviz/nav2_default_view.rviz --ros-args -r __node:=rviz2 -r __node:=rviz2 --params-file /tmp/launch_params_symx73ri'].
[ERROR] [lifecycle_manager-9]: process has died [pid 26149, exit code -6, cmd '/opt/ros/foxy/lib/nav2_lifecycle_manager/lifecycle_manager --ros-args -r __node:=lifecycle_manager_navigation --params-file /tmp/launch_params_1z45enyy --params-file /tmp/launch_params_2wu68eo5 --params-file /tmp/launch_params_cdjg_v_b'].
```

--- a/webots_ros2_driver/CMakeLists.txt
+++ b/webots_ros2_driver/CMakeLists.txt
@@ -168,4 +168,7 @@ ament_export_dependencies(
   tinyxml2_vendor
   TinyXML2
 )
+ament_export_libraries(
+  ${PROJECT_NAME}_imu
+)
 ament_package()

--- a/webots_ros2_driver/CMakeLists.txt
+++ b/webots_ros2_driver/CMakeLists.txt
@@ -136,6 +136,8 @@ install(
   DESTINATION include/webots
 )
 install(TARGETS ${PROJECT_NAME}_imu
+  RUNTIME DESTINATION bin
+  ARCHIVE DESTINATION lib
   LIBRARY DESTINATION lib
 )
 install(

